pkg/prometheus/promcfg.go

```diff
@@ -2281,6 +2282,173 @@ func (cg *ConfigGenerator) generateScrapeConfig(
 			Value: configs,
 		})
 	}
+
+	//ConsulSDConfig
+	if len(sc.Spec.ConsulSDConfigs) > 0 {
+		configs := make([][]yaml.MapItem, len(sc.Spec.ConsulSDConfigs))
+		for i, config := range sc.Spec.ConsulSDConfigs {
+			configs[i] = cg.addBasicAuthToYaml(configs[i], fmt.Sprintf("scrapeconfig/%s/%s/consulsdconfig/%d", sc.Namespace, sc.Name, i), store, config.BasicAuth)
+			configs[i] = cg.addSafeAuthorizationToYaml(configs[i], fmt.Sprintf("scrapeconfig/auth/%s/%s/consulsdconfig/%d", sc.Namespace, sc.Name, i), store, config.Authorization)
+			configs[i] = cg.addOAuth2ToYaml(configs[i], config.Oauth2, store.OAuth2Assets, fmt.Sprintf("scrapeconfig/%s/%s/consulsdconfig/%d", sc.Namespace, sc.Name, i))
+
+			if config.TLSConfig != nil {
+				configs[i] = addSafeTLStoYaml(configs[i], sc.Namespace, *config.TLSConfig)
+			}
+
+			configs[i] = append(configs[i], yaml.MapItem{
+				Key:   "server",
+				Value: config.Server,
+			})
+
+			if config.Token != nil {
+				value, err := store.GetKey(context.Background(), sc.Namespace, monitoringv1.SecretOrConfigMap{
+					Secret: config.Token,
+				})
+
+				if err != nil {
+					level.Error(cg.logger).Log("failed to read %s secret %s", config.Token.Name, jobName)
```

Comment:
this should be checked (and rejected) in `SelectScrapeConfigs()` too.

Reply:
I'm assuming if we're rejecting in the `*ResourceSelector.SelectScrapeConfigs()` I don't need to check for errors, since it's getting from in the memory store.

am I wrong?

Reply:
you're correct :)